crates/cext/src/transpiler/passes/inverse_cancellation.rs

```diff
@@ -0,0 +1,62 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::pointers::mut_ptr_as_ref;
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::converters::dag_to_circuit;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_transpiler::passes::run_inverse_cancellation_standard_gates;
+
+/// @ingroup QkTranspilerPasses
+/// Run the InverseCancellation transpiler pass on a circuit.
+///
+/// Cancels specific gates which are inverses of each other when they occur
+/// back-to-back.
+///
+/// @param circuit A pointer to the circuit to run InverseCancellation on. This circuit
+/// pointed to will be updated with the modified circuit if the pass is able to remove any gates.
+///
+/// # Example
+///
+/// ```c
+///     QkCircuit *qc = qk_circuit_new(2, 2);
+///     uint32_t qargs[1] = {0};
+///     qk_circuit_gate(qc, QkGate_X, qargs, NULL);
+///     qk_circuit_gate(qc, QkGate_H, qargs, NULL);
+///     qk_circuit_gate(qc, QkGate_H, qargs, NULL);
+///     qk_circuit_gate(qc, QkGate_Y, qargs, NULL);
+///     qk_transpiler_pass_standalone_inverse_cancellation(qc);
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``circuit`` is not a valid, non-null pointer to a ``QkCircuit``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_transpiler_pass_standalone_inverse_cancellation(
+    circuit: *mut CircuitData,
+) {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let circuit = unsafe { mut_ptr_as_ref(circuit) };
+    let mut dag = match DAGCircuit::from_circuit_data(circuit, false, None, None, None, None) {
+        Ok(dag) => dag,
+        Err(e) => panic!("{}", e),
```

Comment:
We need to update the other functions by running `to_string()` on a `PyErr` needs python so this will panic while trying to get the error string and say the GIL wasn't available.

```suggestion
        Err(e) => panic!("Internal Circuit -> DAG conversion failed"),
```

Reply:
Done in [bb22aad](https://github.com/Qiskit/qiskit/pull/14840/commits/bb22aad4fadb85f0e0e50e6e8166cbfee0e8f87e).